lib/api/collection_config.rb

```diff
@@ -0,0 +1,120 @@
+module Api
+  class CollectionConfig
+    def self.names_for_feature(product_feature_name)
+      names_for_features[product_feature_name]
+    end
+
+    def self.name_for_klass(resource_klass)
+      Settings.collections.detect do |_, spec|
+        spec[:klass] && spec[:klass].constantize == resource_klass
+      end.try(:first)
+    end
+
+    def self.what_refers_to_feature(product_feature_name)
+      referenced_identifiers[product_feature_name]
+    end
+
+    def self.collections_with_description
+      Settings.collections.each_with_object({}) do |(collection, cspec), result|
+        result[collection] = cspec[:description] if cspec[:options].include?(:collection)
+      end
+    end
+
+    def self.find(collection_name)
+      config = Settings.collections[collection_name.to_sym]
+      new(config) if config
+    end
+
+    def self.names_for_features
+      @names_for_features ||= Settings.collections.each_with_object(Hash.new { |h, k| h[k] = [] }) do |(collection, cspec), result|
+        ident = cspec[:identifier]
+        next unless ident
+        result[ident] << collection
+      end
+    end
+    private_class_method :names_for_features
+
+    def self.referenced_identifiers
+      @referenced_identifiers ||= @cfg.each_with_object({}) do |(collection, cspec), result|
+        next unless cspec[:collection_actions].present?
+        cspec[:collection_actions].each do |method, action_definitions|
+          next unless action_definitions.present?
+          action_definitions.each do |action|
+            identifier = action[:identifier]
+            next if action[:disabled] || result.key?(identifier)
+            result[identifier] = [collection, method, action]
+          end
+        end
+      end
+    end
+    private_class_method :referenced_identifiers
+
+    def initialize(config)
+      @config = config
+    end
+
+    def method_missing(method, *args)
```

Comment:
This forwards any messages to the `Config::Options` object, so it behaves just like that with some extra convenience methods